test/EnvironmentManagerTests/Helper.h

```diff
@@ -0,0 +1,167 @@
+    return GetEnvironmentVariablesFromRegistry(environmentVariablesHKey.get());
+}
+
+inline EnvironmentVariables GetEnvironmentVariablesForProcess()
+{
+    LPWSTR environmentVariablesBlock = GetEnvironmentStrings();
+    VERIFY_IS_NOT_NULL(environmentVariablesBlock);
+
+    LPTSTR variable;
+    variable = (LPTSTR)environmentVariablesBlock;
+
+    StringMap environmentVariables;
+    while (*variable)
+    {
+        std::wstring environmentVariable(variable);
+        size_t locationOfNull = environmentVariable.find_last_of(L'=');
+
+        environmentVariables.Insert(environmentVariable.substr(0, locationOfNull), environmentVariable.substr(locationOfNull + 1));
+
+        variable += lstrlen(variable) + 1;
+    }
+
+    VERIFY_WIN32_BOOL_SUCCEEDED(FreeEnvironmentStrings(environmentVariablesBlock));
+
+    return environmentVariables.GetView();
+}
+
+/// <summary>
+/// Compares two IMapView<winrt::hstring, winrt::hstring> collections for
+/// 1. Have the same size
+/// 2. All the keys in <paramref name="underTest" /> are in <paramref name="real" />
+/// 3. All the values in <paramref name="underTest" /> are in <paramref name="real" />
+/// </summary>
+/// <param name="underTest">The EnvironmentVariables from the WInRT API that is being tested</param>
+/// <param name="real">The Environment Variables that came from GetEnvironmentStrings</param>
+/// <remarks>
+/// This test will alwayse use <paramref name="underTest" /> to compare to <paramref name="real" />.  As long as the size, keys, and values
+/// are compared from <paramref name="underTest" /> to <paramref name="real" /> this will fail in the case that any keys are different, or any values are different
+/// </remarks>
+inline void CompareIMapViews(EnvironmentVariables fromWinRT, EnvironmentVariables fromWin32)
+{
+    std::wstringstream sizeMessage;
+    sizeMessage << L"Size of WinRT Collection: ";
+    sizeMessage << fromWinRT.Size() << "\r\n";
+    sizeMessage << L"Size of Win32 Collection: ";
+    sizeMessage << fromWin32.Size();
+
+    WEX::Logging::Log::Comment(sizeMessage.str().c_str());
+    // Make sure the sizes are the same.
+    VERIFY_ARE_EQUAL(fromWinRT.Size(), fromWin32.Size());
+
+    if (fromWinRT.Size() == 0)
+    {
+        return;
+    }
+
+    auto winRTIterator = fromWinRT.First();
+    do
+    {
+        // Make sure the same key exists in both collections
+        auto winRTKey = winRTIterator.Current().Key();
+
+        std::wstringstream logMessage;
+        logMessage << L"Key from WinRT: ";
+        logMessage << winRTKey.c_str();
+
+        WEX::Logging::Log::Comment(logMessage.str().c_str());
+
+        VERIFY_IS_TRUE(fromWin32.HasKey(winRTKey));
+
+        // Make sure they have the same value
+        auto winRTValue = winRTIterator.Current().Value();
+
+        std::wstringstream valueMessage;
+        valueMessage << L"Value from test: ";
```

Comment:
sprintf FTW!

Not `std::format` until C++20 so [wil::str_printf()](https://github.com/microsoft/wil/wiki/String-helpers#wilstr_printf) is your friend